qiskit/transpiler/_transpiler.py

```diff
@@ -66,144 +65,53 @@ def transpile(circuits, backend, basis_gates=None, coupling_map=None, initial_la
     coupling_map = coupling_map or getattr(backend.configuration(),
                                            'coupling_map', None)
 
-    # step 1: Making the list of dag circuits
-    dags = _circuits_2_dags(circuits)
-
-    # step 2: Transpile all the dags
-
-    # FIXME: Work-around for transpiling multiple circuits with different qreg names.
-    # Make compile take a list of initial_layouts.
-    _initial_layout = initial_layout
-
-    # Pick a good initial layout if coupling_map is not already satisfied
-    # otherwise keep it as q[i]->q[i].
-    # TODO: move this inside mapper pass.
-    initial_layouts = []
-    for dag in dags:
-        if (initial_layout is None and not backend.configuration().simulator
-                and not _matches_coupling_map(dag, coupling_map)):
-            _initial_layout = _pick_best_layout(dag, backend)
-        initial_layouts.append(_initial_layout)
-
-    dags = _dags_2_dags(dags, basis_gates=basis_gates, coupling_map=coupling_map,
-                        initial_layouts=initial_layouts, seed_mapper=seed_mapper,
-                        pass_manager=pass_manager)
-
-    # step 3: Converting the dags back to circuits
-    circuits = _dags_2_circuits(dags)
-
+    circuits = parallel_map(_parallel_transpilation, circuits,
+                            task_args=(backend,),
+                            task_kwargs={'basis_gates': basis_gates,
+                                         'coupling_map': coupling_map,
+                                         'initial_layout': initial_layout,
+                                         'seed_mapper': seed_mapper,
+                                         'pass_manager': pass_manager})
     if return_form_is_single:
         return circuits[0]
     return circuits
 
 
-def _circuits_2_dags(circuits):
-    """Convert a list of circuits into a list of dags.
+def _parallel_transpilation(circuit, backend, basis_gates=None, coupling_map=None,
+                            initial_layout=None, seed_mapper=None,
+                            pass_manager=None):
+    """Perform transpilation of multiple circuits in parallel.
```

Comment:
This function is not transpiling multiple circuits. Just one.

Reply:
yes